admin/class-taxonomy-presenter.php

```diff
@@ -0,0 +1,317 @@
+			$current = 'noindex';
+		}
+
+		$noindex_options['options']            = $this->no_index_options;
+		$noindex_options['options']['default'] = sprintf( $noindex_options['options']['default'], $this->term->taxonomy, $current );
+
+		if ( '0' === get_option( 'blog_public' ) ) {
+			$noindex_options['description'] = '<br /><span class="error-message">' . esc_html__( 'Warning: even though you can set the meta robots setting here, the entire site is set to noindex in the sitewide privacy settings, so these settings won\'t have an effect.', 'wordpress-seo' ) . '</span>';
+		}
+
+		$fields = array(
+			'title' => array(
+				'label'       => __( 'SEO Title', 'wordpress-seo' ),
+				'description' => esc_html__( 'The SEO title is used on the archive page for this term.', 'wordpress-seo' ),
+				'type'        => 'text',
+				'options'     => '',
+			),
+			'desc' => array(
+				'label'       => __( 'SEO Description', 'wordpress-seo' ),
+				'description' => esc_html__( 'The SEO description is used for the meta description on the archive page for this term.', 'wordpress-seo' ),
+				'type'        => 'text',
+				'options'     => '',
+			),
+			'metakey'  => array(
+				'label'       => __( 'Meta keywords', 'wordpress-seo' ),
+				'description' => esc_html__( 'Meta keywords used on the archive page for this term.', 'wordpress-seo' ),
+				'type'        => 'text',
+				'options'     => '',
+				'hide'        => $this->options['usemetakeywords'] !== true,
+			),
+			'canonical'  => array(
+				'label'       => __( 'Canonical', 'wordpress-seo' ),
+				'description' => esc_html__( 'The canonical link is shown on the archive page for this term.', 'wordpress-seo' ),
+				'type'        => 'text',
+				'options'     => '',
+			),
+			'bctitle'  => array(
+				'label'       => __( 'Breadcrumbs title', 'wordpress-seo' ),
+				'description' => sprintf( esc_html__( 'The Breadcrumbs title is used in the breadcrumbs where this %s appears.', 'wordpress-seo' ), $this->term->taxonomy ),
+				'type'        => 'text',
+				'options'     => '',
+				'hide'        => $this->options['breadcrumbs-enable'] !== true,
+			),
+			'noindex'  => array(
+				'label'       => sprintf( __( 'Noindex this %s', 'wordpress-seo' ), $this->term->taxonomy ),
+				'description' => sprintf( esc_html__( 'This %s follows the indexation rules set under Metas and Titles, you can override it here.', 'wordpress-seo' ), $this->term->taxonomy ),
+				'type'        => 'select',
+				'options'     => $noindex_options,
+			),
+			'sitemap_include'  => array(
+				'label'       => sprintf( __( 'Include in sitemap?', 'wordpress-seo' ), $this->term->taxonomy ),
+				'description' => '',
+				'type'        => 'select',
+				'options'     => $this->sitemap_include_options,
+			),
+		);
+
+		unset( $current, $no_index_options );
+
+		return $fields;
+	}
+
+	/**
+	 * Returning the fields for the social media tab
+	 *
+	 * @return array
+	 */
+	public function social_fields() {
+		$social_networks = array(
```

Comment:
I would split this class up in multiple functions to increase readability and enforce single responsibility.